static/src/javascripts/projects/common/modules/article/space-filler.js

```diff
@@ -1,66 +1,47 @@
-/* jscs:disable disallowDanglingUnderscores */
 define([
-    'common/utils/$',
-    'common/utils/fastdom-idle',
+    'common/utils/fastdom-promise',
     'Promise',
     'raven',
     'common/modules/article/spacefinder'
 ], function (
-    $,
-    idleFastdom,
+    fastdom,
     Promise,
     raven,
     spacefinder
 ) {
-    return new SpaceFiller();
+    var lastIteration = Promise.resolve(false);
```

Comment:
We tried to keep our modules as simple as possible, which means doing one thing only. Consequently the surface of our APIs was quite small (and easy to debug). For UI widgets we resorted to a basic plugin architecture that built behaviour on top of the module, again with as few hooks as necessary (mainly creation and destruction). I have to say OO in javascript feels like monkey patching, I tend to use it only for objects with a very short lifespan.